### PR TITLE
fix!: psa pinning APIs - rename requestId to requestid

### LINKED ISSUE #1221: web3 pinning service returns invalid PinStatus object
I am working on https://github.com/ipfs/pinning-services-api-spec/issues/64 and https://github.com/ipfs-shipyard/pinning-service-compliance/issues/4. While attempting to delete all existing pins, I ran into an error with web3.storage's pinning service because the PinResults returned has the property `requestId` instead of `requestid`. 

Check the spec at https://ipfs.github.io/pinning-services-api-spec/#tag/pins/paths/~1pins/get:
image

# Compliance report:

##### Headers

    {
      "Authorization": "Bearer REDACTED"
    }

##### Body

    undefined

##### Headers

    {
      "access-control-allow-origin": "*",
      "access-control-expose-headers": "Link",
      "cf-ray": "6f74509648ab0911-SEA",
      "connection": "close",
      "content-encoding": "gzip",
      "content-type": "application/json;charset=UTF-8",
      "date": "Tue, 05 Apr 2022 18:23:26 GMT",
      "expect-ct": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\"",
      "server": "cloudflare",
      "transfer-encoding": "chunked",
      "vary": "Accept-Encoding"
    }

##### Body

    {
      "count": 3,
      "results": [
        {
          "requestId": "1dfb96af-ab6f-4149-bd4a-e4a043980e40",
          "status": "queued",
          "created": "2022-04-05T17:57:57.774+00:00",
          "pin": {
            "cid": "bafkqadjrgy2dsmjyge2domjsgm2a",
            "_id": "1dfb96af-ab6f-4149-bd4a-e4a043980e40",
            "sourceCid": "bafkqadjrgy2dsmjyge2domjsgm2a",
            "contentCid": "bafkqadjrgy2dsmjyge2domjsgm2a",
            "authKey": REDACTED,
            "name": null,
            "meta": null,
            "deleted": null,
            "created": "2022-04-05T17:57:57.774+00:00",
            "updated": "2022-04-05T17:57:57.774+00:00",
            "pins": [
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWR19qPPiZH4khepNjS3CLXiB7AbrbAD4ZcDjN1UjGUNE1",
                "peerName": "web3-storage-sv15",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWSnniGsyAF663gvHdqhyfJMCjWJv54cGSzcPiEMAfanvU",
                "peerName": "web3-storage-dc13",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWPySxxWQjBgX9Jp6uAHQfVmdq8HG1gVvS1fRawHNSrmqW",
                "peerName": "web3-storage-am6",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWEDMw7oRqQkdCJbyeqS5mUmWGwTp8JJ2tjCzTkHboF6wK",
                "peerName": "web3-storage-sv15-2",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWNuoVEfVLJvU3jWY2zLYjGUaathsecwT19jhByjnbQvkj",
                "peerName": "web3-storage-am6-2",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWKytRAd2ujxhGzaLHKJuje8sVrHXvjGNvHXovpar5KaKQ",
                "peerName": "web3-storage-dc13-2",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWJEfH2MB4RsUoaJPogDPRWbFTi8iehsxsqrQpiJwFNDrP",
                "peerName": "web3-storage-dc13-3",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWRi18oHN1j8McxS9RMnuibcTwxu6VCTYHyLNH2R14qhTy",
                "peerName": "web3-storage-sv15-3",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWQYBPcvxFnnWzPGEx6JuBnrbF1FZq4jTahczuG2teEk1m",
                "peerName": "web3-storage-am6-3",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWDdzN3snjaMJEH9zuq3tjKUFpYHeSGNkiAreF6dQSbCiL",
                "peerName": "web3-storage-am6-4",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWEzCun34s9qpYEnKkG6epx2Ts9oVGRGnzCvM2s2edioLA",
                "peerName": "web3-storage-am6-5",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWHpE5KiQTkqbn8KbU88ZxwJxYJFaqP4mp9Z9bhNPhym9V",
                "peerName": "web3-storage-dc13-4",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWBHvsSSKHeragACma3HUodK5FcPUpXccLu2vHooNsDf9k",
                "peerName": "web3-storage-dc13-5",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWAdxvJCV5KXZ6zveTJmnYGrSzAKuLUKZYkZssLk7UKv4i",
                "peerName": "web3-storage-sv15-5",
                "region": null
              },
              {
                "status": "Unpinned",
                "updated": "2022-04-05T17:57:57.774+00:00",
                "peerId": "12D3KooWKhPb9tSnCqBswVfC5EPE7iSTXhbF4Ywwz2MKg5UCagbr",
                "peerName": "web3-storage-sv15-4",
                "region": null
              }
            ]
          },
          "delegates": []
        },
        {
          "requestId": "817a3ffb-d7ce-4e5b-9a61-d522ce071ea7",
          "status": "pinned",
          "created": "2022-03-29T19:00:54.948+00:00",
          "pin": {
            "cid": "bafkqadjrgy2dqnjyga2diobvge4q",
            "_id": "817a3ffb-d7ce-4e5b-9a61-d522ce071ea7",
            "sourceCid": "bafkqadjrgy2dqnjyga2diobvge4q",
            "contentCid": "bafkqadjrgy2dqnjyga2diobvge4q",
            "authKey": REDACTED,
            "name": null,
            "meta": null,
            "deleted": null,
            "created": "2022-03-29T19:00:54.948+00:00",
            "updated": "2022-03-29T19:00:54.948+00:00",
            "pins": [
              {
                "status": "Pinned",
                "updated": "2022-03-30T22:44:56.649046+00:00",
                "peerId": "12D3KooWJEfH2MB4RsUoaJPogDPRWbFTi8iehsxsqrQpiJwFNDrP",
                "peerName": "web3-storage-dc13-3",
                "region": null
              },
              {
                "status": "Pinned",
                "updated": "2022-03-30T13:45:53.98255+00:00",
                "peerId": "12D3KooWRi18oHN1j8McxS9RMnuibcTwxu6VCTYHyLNH2R14qhTy",
                "peerName": "web3-storage-sv15-3",
                "region": null
              },
              {
                "status": "Pinned",
                "updated": "2022-03-30T19:54:54.839236+00:00",
                "peerId": "12D3KooWQYBPcvxFnnWzPGEx6JuBnrbF1FZq4jTahczuG2teEk1m",
                "peerName": "web3-storage-am6-3",
                "region": null
              }
            ]
          },
          "delegates": []
        },
        {
          "requestId": "52007e1d-a718-4625-b523-9d84db74b0b3",
          "status": "pinned",
          "created": "2022-03-29T18:52:37.675+00:00",
          "pin": {
            "cid": "bafkqadjrgy2dqnjxhe4tkmjtga3a",
            "_id": "52007e1d-a718-4625-b523-9d84db74b0b3",
            "sourceCid": "bafkqadjrgy2dqnjxhe4tkmjtga3a",
            "contentCid": "bafkqadjrgy2dqnjxhe4tkmjtga3a",
            "authKey": REDACTED,
            "name": null,
            "meta": null,
            "deleted": null,
            "created": "2022-03-29T18:52:37.675+00:00",
            "updated": "2022-03-29T18:52:37.675+00:00",
            "pins": [
              {
                "status": "Pinned",
                "updated": "2022-03-30T13:26:26.782542+00:00",
                "peerId": "12D3KooWJEfH2MB4RsUoaJPogDPRWbFTi8iehsxsqrQpiJwFNDrP",
                "peerName": "web3-storage-dc13-3",
                "region": null
              },
              {
                "status": "Pinned",
                "updated": "2022-03-30T16:19:20.628727+00:00",
                "peerId": "12D3KooWRi18oHN1j8McxS9RMnuibcTwxu6VCTYHyLNH2R14qhTy",
                "peerName": "web3-storage-sv15-3",
                "region": null
              },
              {
                "status": "Pinned",
                "updated": "2022-03-31T03:59:37.478391+00:00",
                "peerId": "12D3KooWQYBPcvxFnnWzPGEx6JuBnrbF1FZq4jTahczuG2teEk1m",
                "peerName": "web3-storage-am6-3",
                "region": null
              }
            ]
          },
          "delegates": []
        }
      ]
    }

### PR DESCRIPTION
**Pull request by flea89**
resolves #1221 

## Notes
While it’s doubtless the API spec specifies the `PinStatus` should have `requestid` and not `requestId` I wonder why, when using `go-ipfs`  to add a remote pin with the current implementation, it doesn’t complain/blow up because of that.
In fact, it does behave as expected.

I wonder before merging this we should be checking `ipfs-go` implementation to make sure it's not expecting `requestId` id capital `I`?

### CODE REVIEW

File: packages/api/test/pin.spec.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ const createPinRequest = async (cid, token) => {
   })).json()
 }
 
-describe('Pinning APIs endpoints', () => {
+describe.only('Pinning APIs endpoints', () => {
```
Comment (alanshaw) [inline packages/api/test/pin.spec.js:81]: Please remove

**Review by alanshaw — CHANGES_REQUESTED** [review]

File: packages/api/test/pin.spec.js (hunk 1 above)
Comment (flea89) [reply packages/api/test/pin.spec.js:81]: 😓
Sorry about that, so annoying.
I wonder if we should consider adding a step in the CI to catch these things.

**Comment by alanshaw** [comment]
> @alanshaw any thoughts on my note in the [description](https://github.com/web3-storage/web3.storage/pull/1267#issue-1212524710)?

It is expecting `requestid`: https://github.com/ipfs/go-pinning-service-http-client/blob/9087f9a0e9b39bafb151c73cd21d6d787450a0e4/openapi/model_pin_status.go#L20

File: packages/api/src/pins.js [hunk 2]
```diff
@@ -192,7 +192,7 @@ export async function pinsGet (request, env, ctx) {
  */
 function getPinStatus (pinRequest) {
```
Comment (alanshaw) [inline packages/api/src/pins.js:193]: Nit: I'd have named this something like `toPinStatusResponse` - this isn't _retrieving_ anything it's transforming something.

**Review by alanshaw — APPROVED** [review]
LGTM

File: packages/api/src/pins.js (hunk 2 above)
Comment (flea89) [reply packages/api/src/pins.js:193]: Agree, I'll change it.

**Pull request merged**